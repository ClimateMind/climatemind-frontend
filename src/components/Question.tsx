import React, { useState } from 'react';
import {
  Grid,
  makeStyles,
  FormControl,
  FormControlLabel,
  FormLabel,
  RadioGroup,
} from '@material-ui/core';
import GreenRadio from './GreenRadio';
import { TAnswers } from '../types/types';
import Paragraphs from './Paragraphs';

type Props = {
  questionId: number; //Identify the question
  questionNumber: number; //Q1,Q2,Q3 Etc
  question: string;
  answers: TAnswers;
  setAnswer: (questionId: number, value: string) => void;
  isSmall: boolean;
};

const styles = makeStyles({
  root: {
    flexGrow: 1,
    width: '100%',
    padding: 0,
    margin: 0,
  },
  answers: {},
  formControl: {
    padding: '4px 0 !important',
    margin: '0 -8px 0 0',
  },
  formControlLargeScreen: {
    padding: '14px 0 !important',
  },
  radioLargeScreen: {
    marginRight: '40px',
  },
  questionHeader: {
    margin: '1em 0',
    width: '100%',
    minHeight: '150px',
    display: 'block',
  },
  questionHeaderLargeScreen: {
    marginTop: '64px',
    marginBottom: '1em',
    width: '100%',
    minHeight: '105px',
    display: 'block',
  },
  questionNumber: {
    marginRight: '1em',
  },
  answerContainer: {
    display: 'flex',
    '& > *': {
      display: 'inline-block',
    },
  },
});

const Question: React.FC<Props> = ({
  question,
  answers,
  questionId,
  setAnswer,
  isSmall
}) => {
  const classes = styles();

  const [choosenAnswer, setChoosenAnswer] = useState(''); //Input Control

  // Controlled Input - Handle when the user picks an answer
  const handleAnswer = (e: React.ChangeEvent<HTMLInputElement>) => {
    const choosenAnswer = e.target.value;
    setChoosenAnswer(e.target.value);
    // Set Input State
    setTimeout(() => {
      setAnswer(questionId, choosenAnswer);
    }, 400);
  };

  return (
    <div className={classes.root} data-testid="Question">
      <FormControl component="fieldset">
        <Grid container>
          <Grid item xs={12}>
            <FormLabel
              component="legend"
              className={isSmall ? classes.questionHeader : classes.questionHeaderLargeScreen}
              id="questionText"
            >
              <Paragraphs text={question} fontSize="18px" bold />
            </FormLabel>
          </Grid>

          {/* Question - Answer Text and Radios */}
         <Grid item xs={12}>
            <RadioGroup
              aria-label="question"
              name={question}
              value={choosenAnswer}
              onChange={(e) => handleAnswer(e)}
            >
              {answers.map((answer) => {
                return (
                  <FormControlLabel
                    className={isSmall ? classes.formControl : classes.formControlLargeScreen}
                    value={`${answer.id}`}
                    key={answer.id}
                    control={<GreenRadio className={isSmall ? '' : classes.radioLargeScreen} color="secondary" />}
                    label={answer.text}
                    labelPlacement={isSmall ? "start" : "end"}
                  />
                );
              })}
            </RadioGroup> 
          </Grid> 
        </Grid>
      </FormControl>
    </div>
  );
};

export default Question;
