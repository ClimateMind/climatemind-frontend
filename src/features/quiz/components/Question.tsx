import { useState } from 'react';
import { Grid, FormControl, FormControlLabel, FormLabel, RadioGroup, Radio } from '@mui/material';
import { TAnswers } from '../../../types/types';

import { CmTypography } from 'shared/components';

type Props = {
  questionId: number; //Identify the question
  questionNumber: number; //Q1,Q2,Q3 Etc
  question: string;
  answers: TAnswers;
  setAnswer: (questionId: number, value: string) => void;
  isSmall: boolean;
};

const Question: React.FC<Props> = ({
  question,
  answers,
  questionId,
  setAnswer,
  isSmall,
}) => {
  const [choosenAnswer, setChoosenAnswer] = useState(''); //Input Control

  // Controlled Input - Handle when the user picks an answer
  const handleAnswer = (e: React.ChangeEvent<HTMLInputElement>) => {
    const choosenAnswer = e.target.value;
    setChoosenAnswer(e.target.value);

    // Scroll to top after each question
    window.scrollTo(0, 0);

    // Set Input State
    setTimeout(() => {
      setAnswer(questionId, choosenAnswer);
    }, 400);
  };

  return (
    <div style={styles.root} data-testid="Question">
      <FormControl component="fieldset">
        <Grid container>
          <Grid item xs={12}>
            <FormLabel
              component="legend"
              style={
                isSmall
                  ? styles.questionHeader
                  : styles.questionHeaderLargeScreen
              }
              id="questionText"
            >
              {question.split('\n\n').map((p, i) => (
                <CmTypography variant='body' style={{ fontSize: '18px', fontWeight: 'bold' }} key={i}>
                  {p}
                </CmTypography>
              ))}
            </FormLabel>
          </Grid>

          {/* Question - Answer Text and Radios */}
          <Grid item xs={12}>
            <RadioGroup
              aria-label="question"
              name={question}
              value={choosenAnswer}
              onChange={(e) => handleAnswer(e)}
            >
              {answers.map((answer) => {
                return (
                  <FormControlLabel
                    style={
                      isSmall
                        ? styles.formControl
                        : styles.formControlLargeScreen
                    }
                    value={`${answer.id}`}
                    key={answer.id}
                    control={
                      <Radio style={{ color: '#39F5AD', marginRight: isSmall ? 0 : 40 }} />
                    }
                    label={answer.text}
                    labelPlacement={isSmall ? 'start' : 'end'}
                  />
                );
              })}
            </RadioGroup>
          </Grid>
        </Grid>
      </FormControl>
    </div>
  );
};

const styles: { [key: string]: React.CSSProperties } = {
  root: {
    flexGrow: 1,
    width: '100%',
    padding: 0,
    margin: 0,
  },
  formControl: {
    padding: '4px 0 !important',
    margin: '0 -8px 0 0',
  },
  formControlLargeScreen: {
    padding: '14px 0 !important',
  },
  questionHeader: {
    margin: '1em 0',
    width: '100%',
    minHeight: '100px',
    display: 'block',
  },
  questionHeaderLargeScreen: {
    marginTop: '64px',
    marginBottom: '1em',
    width: '100%',
    minHeight: '80px',
    display: 'block',
  },
  questionNumber: {
    marginRight: '1em',
  },
};

export default Question;
