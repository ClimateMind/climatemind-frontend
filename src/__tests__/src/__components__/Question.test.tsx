import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Question from '../../../components/Question';

const TestQuestion = {
  questionId: 1,
  questionNumber: 1,
  question: 'What is the capital of spain',
  answers: ['Madrid', 'Seville', 'London', 'Alicante', 'Marbella'],
};

const setAnswer = jest.fn();

describe('Question Renders', () => {
  it('it loads', () => {
    const { getByTestId } = render(
      <Question {...TestQuestion} setAnswer={setAnswer} />
    );
    expect(getByTestId('Question')).toBeInTheDocument();
  });
  it('Displays the question', () => {
    const { getByText } = render(
      <Question {...TestQuestion} setAnswer={setAnswer} />
    );
    expect(getByText('What is the capital of spain')).toBeInTheDocument();
  });
  it('displays all of answers', () => {
    const { getByText } = render(
      <Question {...TestQuestion} setAnswer={setAnswer} />
    );
    expect(getByText('Madrid')).toBeInTheDocument();
    expect(getByText('Seville')).toBeInTheDocument();
    expect(getByText('London')).toBeInTheDocument();
    expect(getByText('Alicante')).toBeInTheDocument();
    expect(getByText('Marbella')).toBeInTheDocument();
  });

  it('it sets the answer correctly', () => {
    const { getByText } = render(
      <Question {...TestQuestion} setAnswer={setAnswer} />
    );
    const myAnswer = getByText('Madrid');
    fireEvent.click(myAnswer);
  });
});
