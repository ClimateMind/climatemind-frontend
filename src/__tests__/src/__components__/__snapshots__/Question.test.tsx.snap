// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Renders matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-root-46 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
        data-testid="Question"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
        >
          <fieldset
            class="MuiFormControl-root"
          >
            <legend
              class="MuiFormLabel-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <div
                    class="MuiBox-root MuiBox-root-47"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                    >
                      Q
                      1
                    </h4>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5"
                  >
                    What is the capital of spain
                  </h5>
                </div>
              </div>
            </legend>
            <div
              aria-label="question"
              class="MuiFormGroup-root"
              role="radiogroup"
            >
              <div
                class="MuiBox-root MuiBox-root-48"
              >
                <div
                  class="MuiBox-root MuiBox-root-49"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-53"
                          name="What is the capital of spain"
                          type="radio"
                          value="0"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-54"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Madrid
                    </span>
                  </label>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-57"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-53"
                          name="What is the capital of spain"
                          type="radio"
                          value="1"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-54"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Seville
                    </span>
                  </label>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-58"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-53"
                          name="What is the capital of spain"
                          type="radio"
                          value="2"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-54"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      London
                    </span>
                  </label>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-59"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-53"
                          name="What is the capital of spain"
                          type="radio"
                          value="3"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-54"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Alicante
                    </span>
                  </label>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-60"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-53"
                          name="What is the capital of spain"
                          type="radio"
                          value="4"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-54"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Marbella
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-root-46 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      data-testid="Question"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <fieldset
          class="MuiFormControl-root"
        >
          <legend
            class="MuiFormLabel-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
              >
                <div
                  class="MuiBox-root MuiBox-root-47"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    Q
                    1
                  </h4>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5"
                >
                  What is the capital of spain
                </h5>
              </div>
            </div>
          </legend>
          <div
            aria-label="question"
            class="MuiFormGroup-root"
            role="radiogroup"
          >
            <div
              class="MuiBox-root MuiBox-root-48"
            >
              <div
                class="MuiBox-root MuiBox-root-49"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-53"
                        name="What is the capital of spain"
                        type="radio"
                        value="0"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Madrid
                  </span>
                </label>
              </div>
              <div
                class="MuiBox-root MuiBox-root-57"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-53"
                        name="What is the capital of spain"
                        type="radio"
                        value="1"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Seville
                  </span>
                </label>
              </div>
              <div
                class="MuiBox-root MuiBox-root-58"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-53"
                        name="What is the capital of spain"
                        type="radio"
                        value="2"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    London
                  </span>
                </label>
              </div>
              <div
                class="MuiBox-root MuiBox-root-59"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-53"
                        name="What is the capital of spain"
                        type="radio"
                        value="3"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Alicante
                  </span>
                </label>
              </div>
              <div
                class="MuiBox-root MuiBox-root-60"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-50 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-53"
                        name="What is the capital of spain"
                        type="radio"
                        value="4"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Marbella
                  </span>
                </label>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
